parley/src/layout/cursor.rs

```diff
@@ -373,6 +395,27 @@ impl Cursor {
     fn visual_order_key(&self) -> (usize, f32) {
         (self.path.line_index(), self.visual_offset)
     }
+
+    #[cfg(feature = "accesskit")]
+    pub fn to_access_position<B: Brush>(
+        &self,
+        layout: &Layout<B>,
+        layout_access: &LayoutAccessibility,
+    ) -> Option<TextPosition> {
+        let run_path = (self.path.line_index(), self.path.run_index());
+        let id = layout_access.access_ids_by_run_path.get(&run_path)?;
+        let mut character_index = self.path.logical_index();
+        if self.affinity == Affinity::Upstream {
```

Comment:
This could probably do with a comment. I'm finding it hard to reason about what this is doing.

Part of that is how confusing Upstream/Downstream are, which isn't your fault.

Reply:
I hope that's clear now without being too verbose.

Reply:
Thank yeah, that really helps!